packages/block-editor/src/components/global-styles/shadow-panel-components.js

```diff
@@ -149,3 +165,30 @@ function renderShadowToggle() {
 		);
 	};
 }
+
+export function useShadowPresets( settings ) {
+	return useMemo( () => {
+		if ( ! settings?.shadow ) {
+			return EMPTY_ARRAY;
+		}
+
+		const defaultPresetsEnabled = settings?.shadow?.defaultPresets;
+		const { default: defaultShadows, theme: themeShadows } =
+			settings?.shadow?.presets;
```

Comment:
```suggestion
			settings?.shadow?.presets ?? {};
```

This should prevent a critical error that occurs when shadows are enabled _only_ on certain blocks.

See the screencast below.

https://github.com/WordPress/gutenberg/assets/54422211/e7aade5c-04f6-49e7-94f1-515c9257df7d

Reply:
Great catch. Thank you!